File: packages/appwsgi/appwsgi%201001.tar.bz2/appwsgi/www/query/query.py
# Copyright(c) gert.cuykens@gmail.com
from frame import Frame
from db import Db

def application(environ, response):
    document = environ['wsgi.input'].read(int(environ.get('CONTENT_LENGTH','0')))
    v = Frame.read(document)
    db = Db('localhost',v['user'],v['password'],v['database'])
    db.execute(v['sql'],())
    xml = '<?xml version="1.0" encoding="UTF-8"?>\n<root>\n'
    xml+= ' <sql>'+Db.escape(v['sql'])+'</sql>\n'
    xml+= Frame.write(db)
    xml+= '</root>'
    response('200 OK', [('Content-type', 'text/xml')])
    return [xml]

File: packages/appwsgi/appwsgi%20714.tar.bz2/trunk/www/query/query.js
// Copyright(c) gert.cuykens@gmail.com
query=
{ 

 'submit':function()
 {
  var server=document.getElementById('server').value
  var user=document.getElementById('user').value
  var password=document.getElementById('password').value
  var database=document.getElementById('database').value
  var sql=document.getElementById('sql').value
  sql=sql+''
  sql=sql.replace(/&/g,'&amp;')
  sql=sql.replace(/</g,'&lt;')
  sql=sql.replace(/>/g,'&gt;')
  http.xml ='<?xml version="1.0" encoding="UTF-8" ?>\n'
  http.xml+='<root>\n'
  http.xml+=' <server>'+server+'</server>\n'
  http.xml+=' <user>'+user+'</user>\n'
  http.xml+=' <password>'+password+'</password>\n'
  http.xml+=' <database>'+database+'</database>\n'
  http.xml+=' <sql>'+sql+'</sql>\n'
  http.xml+='</root>'
  http.fml=function(xml)
  {
   var v
   if ( v=document.getElementById('result') ) v.parentNode.removeChild(v)
   result={'id':'result','onselect':function(){}}
   dg.init(result,xml)
  }
  http.url='query.py'
  http.send()
 }

}
